source/extensions/transport_sockets/tls/ssl_socket.h

```diff
@@ -89,6 +94,9 @@ class SslSocket : public Network::TransportSocket,
   mutable std::string cached_sha_256_peer_certificate_digest_;
   mutable std::string cached_url_encoded_pem_encoded_peer_certificate_;
   mutable std::string cached_url_encoded_pem_encoded_peer_cert_chain_;
+  Envoy::Ssl::PrivateKeyOperationsProviderSharedPtr provider_;
+  Envoy::Ssl::PrivateKeyOperationsPtr ops_;
```

Comment:
Do we need to store this?

Reply:
The provider is no longer needed. The `ops_` (or actually a vector now) is meant for connecting the operations lifecycle with the `SslSocket` lifecycle.